### PR TITLE
Port debounce fix from cujojs/most#515

### PR DESCRIPTION
**Pull request by briancavalier**
Post debounce fix from cujojs/most#515

See also https://github.com/cujojs/most/issues/514

### Todo

- [x] Unit test

### CODE REVIEW

File: packages/core/test/limit-test.js [hunk 1]
```diff
@@ -65,6 +69,16 @@ describe('debounce', function () {
     })
   })
 
+  it('should not repeat last event when end is delayed', () => {
+    // See https://github.com/cujojs/most/issues/514
+    var expected = Math.random()
+    var end = delay(5, now(null))
+    var debounced = debounce(1, until(end, startWith(expected, never())))
+
+    return collectEventsFor(6, debounced)
```
Comment (Frikki) [inline packages/core/test/limit-test.js]: Why 6 ticks?

Comment (briancavalier) [reply packages/core/test/limit-test.js]: Lol, I think I was playing around with the test, and forgot to change it back to 5.

Also, `var` 😲

Comment (briancavalier) [reply packages/core/test/limit-test.js]: See ce1067b

Comment (Frikki) [reply packages/core/test/limit-test.js]: I didn't even notice var 😮

**Pull request merged**